app/services/new_statement_classifier.rb

```diff
@@ -0,0 +1,169 @@
+  end
+
+  def done
+    classified_statements.fetch(:done, [])
+  end
+
+  def reverted
+    classified_statements.fetch(:reverted, [])
+  end
+
+  def removed
+    classified_statements.fetch(:removed, [])
+  end
+
+  def to_a
+    statements.to_a
+              .map { |s| decorate_statement(s) }
+              .select(&:type) # remove statements without a type
+  end
+
+  private
+
+  def classified_statements
+    @classified_statements ||= to_a.each_with_object({}) do |statement, h|
+      h[statement.type] ||= []
+      h[statement.type] << statement
+    end
+  end
+
+  def statement_type(statement)
+    if statement.removed_from_source? && !statement.done_or_reverted?
+      nil
+    elsif statement.removed_from_source? && statement.done_or_reverted?
+      :removed
+    elsif statement.unverifiable?
+      :unverifiable
+    elsif statement.done?
+      :done
+    elsif statement.reverted?
+      :reverted
+    elsif statement.manually_actionable?
+      :manually_actionable
+    elsif statement.actionable?
+      :actionable
+    elsif statement.verified?
+      :reconcilable
+    else
+      :verifiable
+    end
+  end
+
+  def decorate_statement(statement)
+    comparison = comparison_for_statement(statement)
+    NewStatementDecorator.new(statement, comparison).tap do |s|
+      s.type = statement_type(s)
+    end
+  end
+
+  def person_item_from_transaction_id
+    return unless transaction_id
+    statements.first.person_item
+  end
+
+  def position_held_data
+    @position_held_data ||= NewRetrievePositionData.run(
+      page.position_held_item,
+      page.parliamentary_term_item,
+      person_item_from_transaction_id
+    )
+  end
+
+  def comparison_for_statement(statement)
+    MembershipComparison.new(
+      existing:      existing_statements(statement),
+      suggestion:    mapped_statement(statement),
+      require_party: false
+    )
+  end
+
+  def merged_then_deleted(data)
+    data.merged_then_deleted.split.map { |item| item.split('/').last }
+  end
+
+  def existing_statements(statement)
+    position_held_data.each_with_object({}) do |data, memo|
+      person_items = [data.person] + merged_then_deleted(data)
+      next unless person_items.include?(statement.person_item)
+
+      memo[data.position] = {
+        position: { id: page.position_held_item },
+        start:    data.position_start,
+        end:      data.position_end,
+        term:     {
+          id:    data.term,
+          start: data.term_start,
+          end:   data.term_end,
+        },
+        party:    { id: data.group },
+        district: { id: data.district },
+        data:     {
+          statement_uuid: data.position,
+          revision:       data.revision,
+        },
+      }
+    end
+  end
+
+  def mapped_statement(statement)
+    {
+      position: { id: page.position_held_item },
+      term:     {
+        id:    page.parliamentary_term_item,
+        start: '2015-12-03', # FIXME
+        end:   nil, # FIXME
```

Comment:
Do these need to be fixed before this is merged?

Reply:
Fixed in 3e308cba580efa5ed987c8f547d60c0ff0e50123